src/Compilers/Test/Utilities/CSharp/Extensions.cs

```diff
@@ -163,25 +163,52 @@ private static ImmutableArray<string> SplitMemberName(string name)
         }
 
         public static Symbol GetMember(this CSharpCompilation compilation, string qualifiedName)
+            => compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static ISymbol GetMember(this Compilation compilation, string qualifiedName)
+            => compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static T GetMember<T>(this CSharpCompilation compilation, string qualifiedName) where T : Symbol
+            => (T)compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static T GetMember<T>(this Compilation compilation, string qualifiedName) where T : ISymbol
+            => (T)compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static IMethodSymbol GetCopyConstructor(this Compilation compilation, string qualifiedTypeName)
```

Comment:
>GetCopyConstructor [](http://example.com/codeflow?start=36&length=18)

Only record classes have copy constructors. Should that be incorporated into the logic?

Reply:
I don't think that's necessary. The main property of a copy constructor is that it takes another instance of the same type.

Reply:
> The main property of a copy constructor is that it takes another instance of the same type.

Semantics of a constructor like that is not prescribed/defined for types other than record classes. It is not appropriate to assume that the constructor has copy semantics otherwise.